pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -346,6 +347,139 @@ However, as blocks are currently executed sequentially, the
 `Building executable` block should finish in 5 hours minus the time it took the
 `Creating Docker Image` block to finish.
 
+## fail_fast
+
+The `fail_fast` property enables you to set a desired policy for behavior of
+other blocks when one of them fails.
+
+If all blocks are run sequentially, this is not necessary because if one block
+fails, the ones in a sequence after it won't be started and whole pipeline will
+fail immediately.
+
+However, if you have blocks in your pipeline running in parallel (this can be
+achieved by setting [dependencies](#dependencies-in-blocks) property for each
+block), when one block fails, others which do not directly depend on it will
+continue to execute and use time and resources even though whole pipeline will
+eventually fail.
+
+To optimize this, you can choose to set either `cancel` or `stop` fail_fast
```

Comment:
s/fast_fail/fast failing

Reply:
We had a meeting with @darkofabijan and @shiroyasha where we decided to go with `fail_fast` as a yaml definition property instead of `fast_fail` or `fast_failing`.

Apparently the `fail fast` is the term most of industry uses and it is what you need to search for to find similar behaviors on other services and what will new users most probably look for.

At same time we found that search for `fast failing` only yields Semaphore Classic docs  as a result and so it might be an improperly coined term or a term which is not intuitive for native English speakers.

The other thing I'm not sure here is should we use `fail_fast` or `fail fast` when talking about general behavior policy instead of yaml definition property? 

I used same form everywhere for consistency but that might not be correct.

Reply:
OK. My comment was mostly about your last point, using a human phrase when describing the behavior instead of the YAML keyword. I'll let you pick the best one.